### ISSUE TITLE
how to use focal loss in yolov9?

### ISSUE THREAD

**Issue by ridvanozdemir** [WongKinYiu/yolov9#182]
Is it possible to use focal loss as a loss function in yolov9?

**Comment by WongKinYiu** [comment]
https://github.com/WongKinYiu/yolov9/blob/main/utils/loss_tal_dual.py#L118

set `fl_gamma` in `hyp.yaml`.

**Comment by ridvanozdemir** [comment]
you mean in this file:
https://github.com/WongKinYiu/yolov9/blob/main/data/hyps/hyp.scratch-high.yaml

after I set  ` fl_gamma ` ,  Can I also modify focal loss function? I want to set different alpha values according to classes ? ( I want to assign different alpha values to classes according to the ratio of the number of samples in the training dataset. )

**Comment by WongKinYiu** [comment]
https://github.com/WongKinYiu/yolov9/blob/main/utils/loss_tal_dual.py#L42

You need modify forward function of focal loss to enable dynamic alpha.

**Comment by ridvanozdemir** [comment]
hi again!

how can I access class information from loss_tal_dual.py file? I mean for exmaple I have 5 classes, and I want to implement alpha=0.375 for class 3 and alpha will remain 0.25 for the rest. Can you help me to modify forward function of focal loss like this?

**Comment by WongKinYiu** [comment]
for example, you have 8 samples, their corresponding gt labels are:
`[1, 1, 2, 3, 5, 3, 4, 3]`

create alpha list:
`[.25, .25, .25, .375, .25, .375, .25, .375]`

modify forward function of focal loss:
`def forward(self, pred, true, alpha):`
replace `self.alpha` in forward to `alpha`.

**Comment by ridvanozdemir** [comment]
thank you!
